com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl

```diff
@@ -323,7 +323,17 @@ void GetPunctualLightVectors(float3 positionWS, LightData light, out float3 L, o
     if (light.lightType == GPULIGHTTYPE_PROJECTOR_BOX)
     {
         L = -light.forward;
-        distances.xyz = 1; // No distance or angle attenuation
+
+        if (light.rangeAttenuationBias == 1.0) // Light uses range attenuation
+        {
+            float dist    = -dot(lightToSample, L);
+            float distSq  = dist * dist;
+            float distRcp = rcp(dist);
```

Comment:
mistake here, sohuld be rcp(distSq  )

Reply:
Looking at the code, it seems that pyramid light don't support the rangeAttenuationBias  either? guess we have miss something here, we should unify all this. Pyramid and box should both have the option rangeAttenuationBias  , only diff is that pyramid have it by default and box not.

then we can unify the code:

void GetPunctualLightVectors(float3 positionWS, LightData light, out float3 L, out float4 distances)
{
    float3 lightToSample = positionWS - light.positionRWS;
    distances.w = dot(lightToSample, light.forward);

      float3 unL     = -lightToSample;
      float  distRcp = 1.0;

  if (light.rangeAttenuationBias == 1.0)
{
        float  distSq  = dot(unL, unL);	
        distRcp = rsqrt(distSq);
        float  dist    = distSq * distRcp;
      distances.xyz = float3(dist, distSq, distRcp);
}

L = light.lightType == GPULIGHTTYPE_PROJECTOR_BOX ? -light.forward : unL * distRcp;
}

Reply:
Isn't rcp(dist) correct since the other branch does rsqrt(distSq) ? And the variable is named distRcp so it should be renamed if thats a mistake

Reply:
The pyramid does support rangeAttenuationBias.
But I don't think it is possible to refactor the code, plus they don't have much in common appart from the ModifyDistancesForFillLighting call

Reply:
indeed you are right, ok let's do this :)